### PR TITLE
R2S Config/Step1

### LINKED ISSUE #23: Prepare R2S configuration and/or model for partial activation
R2S allows for only some materials to be activated and contribute to the photon source.

This may be as simple as removing some cells from the goemetry.

### PR DESCRIPTION
**Pull request by anu1217**
fixes #23

### CODE REVIEW

File: WC_Layers/Conversion_h5m.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+mb = core.Core()
+mb.load_file('OpenMC_Mesh.h5m')
+
+# Retrieve the tally
+tally = sp.get_tally(id=2)
+
+# Load the mesh
+meshes = sp.meshes
+mesh = meshes[1]
```
Comment (gonuke) [inline WC_Layers/Conversion_h5m.py]:
```suggestion
mesh = sp.meshes[1]
```

File: WC_Layers/Conversion_h5m.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+tally_data = tally.get_reshaped_data(value='mean')
+
+#Summing over cell filter:
+flux_sum_en = tally_data.sum(axis=0)
```
Comment (gonuke) [inline WC_Layers/Conversion_h5m.py]:
```suggestion
flux_spectra_on_mesh = tally_data.sum(axis=0)
```

File: WC_Layers/Conversion_h5m.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+plt.xlabel('Energy [eV]')
+plt.ylabel('Flux [n/cm^2-s]')
+plt.loglog(ebounds[:,0],flux_sum_en[0,:,0,0])
+plt.savefig("Flux_Graph.png")
+
+# get all tets from the MOAB mesh
+all_tets = mb.get_entities_by_type(0, types.MBTET)
+n_flux_tag = mb.tag_get_handle('FLUX_MESH', 175, types.MB_TYPE_DOUBLE, types.MB_TAG_DENSE, create_if_missing=True)
+mb.tag_set_data(n_flux_tag, all_tets, flux_sum_en[:,:,0,0]) # the shape of the data will need to be checked here
```
Comment (gonuke) [inline WC_Layers/Conversion_h5m.py]:
```suggestion
mb.tag_set_data(n_flux_tag, all_tets, flux_spectra_on_mesh[:,:,0,0]) # the shape of the data will need to be checked here
```

File: WC_Layers/config.ini [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+structured: False
+# Specify whether this problem uses sub-voxel r2s
+sub_voxel: False
+
+[step1]
+# Path to MCNP MESHTAL file containing neutron fluxes or a DAG-MCNP5
+# unstructured mesh tally .h5m file.
+meshtal: Mesh_with_Tally.h5m
+# Tally number within the meshtal file containing the fluxes for activation.
+tally_num: 2
```
Comment (gonuke) [inline WC_Layers/config.ini:12]: Is this relevant for an unstructured mesh?  It doesn't seem like your script to convert from OpenMC to H5M preserves any tally number (which is fine if we don't actually need it).

File: WC_Layers/config.ini [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+# The name of the tag used to store flux data on the mesh. For unstructured
+# mesh this tag must already exist within the file specified in <meshtal>.
+flux_tag: FLUX_MESH
+# Path to the DAGMC material-laden geometry file (.h5m).
+geom: DAGMC_Geometry.h5m
```
Comment (gonuke) [inline WC_Layers/config.ini:17]: Does this geometry have materials in it?  Where is the script that does that?

**Review by gonuke — CHANGES_REQUESTED** [review]
Thanks for figuring a lot of this out.  A few comments here, and curious if you have any evidence of it working?

File: WC_Layers/config.ini (hunk 5 above)
Comment (anu1217) [reply WC_Layers/config.ini:17]: Yes, this is material laden. PyNE_Lib.py creates a PyNE material library which is then assigned to the DAG-MC geometry using the uwuw_preproc tool (done inside a terminal).

File: WC_Layers/config.ini (hunk 4 above)
Comment (anu1217) [reply WC_Layers/config.ini:12]: No, it's not relevant here--only data from one tally is written to h5m, so the same ALARA input file is printed regardless of which number is chosen here

**Comment by anu1217** [comment]
I'd say that this is working in the sense that it is creating a photon source distribution (as in #26), but the source printed at each decay time (source_1.h5m & source_2.h5m) is the same, so this might be something to look into

**Review by gonuke — APPROVED** [review]
Thanks for figuring some of this out @anu1217

**Pull request merged**